jupyterhub/spawner.py

```diff
@@ -306,6 +307,30 @@ def _default_access_scopes(self):
             f"access:servers!user={self.user.name}",
         ]
 
+    group_overrides = Union(
+        [Callable(), Dict()],
+        help="""
+        Override specific traitlets based on group membership of the user.
+
+        This can be a dict, or a callable that returns a dict. The keys of the dict
+        are *only* used for lexicographical sorting, to guarantee consistent
+        ordering of the overrides. If it is a callable, it may be async, and will
+        be passed one parameter - the spawner instance. It should return a dictionary.
+
+        The value of the dict is a dict, with the following keys:
+
```

Comment:
This behaviour is easy to understand if you're already familiar with how Z2JH and KubeSpawner are configured, but we're now introducing this to a new audience.

I think an example configuration, either here or elsewhere in the JupyterHub documentation, would be useful.

Reply:
@manics added one! what do you think?